### PR TITLE
FORGE-2744: extracted some EE interfaces from commands.

### PR DESCRIPTION
**Pull request by metacosm**

### CODE REVIEW

**Comment by gastaldi** [comment]
Add to whitelist

**Comment by gastaldi** [comment]
ok to test

File: javaee/impl/src/main/java/org/jboss/forge/addon/javaee/rest/ui/RestNewEndpointCommandImpl.java [hunk 1]
```diff
@@ -121,4 +121,14 @@ public JavaClassSource decorateSource(UIExecutionContext context, Project projec
 
       return source;
    }
+
+   @Override
+   public UISelectMany<RestMethod> getMethods() {
```
Comment (gastaldi) [inline javaee/impl/src/main/java/org/jboss/forge/addon/javaee/rest/ui/RestNewEndpointCommandImpl.java]: It looks like the Forge formatter is not being used in here? https://github.com/forge/core/blob/master/eclipse-code-formatter-profile.xml

File: javaee/tests/src/test/java/org/jboss/forge/addon/javaee/cdi/CDIOperationsTest.java [hunk 2]
```diff
@@ -94,7 +92,7 @@ public void testGetProjectInjectableBeans() throws Exception
       }
       Assert.assertEquals(1, cdiOperations.getProjectInjectableBeans(project).size());
 
-      try (CommandController controller = uiTestHarness.createCommandController(JPANewEntityCommand.class,
+      try (CommandController controller = uiTestHarness.createCommandController(JPANewEntityCommandImpl.class,
```
Comment (gastaldi) [inline javaee/tests/src/test/java/org/jboss/forge/addon/javaee/cdi/CDIOperationsTest.java]: Do the tests need to change?

Comment (metacosm) [inline javaee/tests/src/test/java/org/jboss/forge/addon/javaee/cdi/CDIOperationsTest.java]: Not sure, actually… :(

File: javaee/impl/src/main/java/org/jboss/forge/addon/javaee/rest/ui/RestNewEndpointCommandImpl.java (hunk 1 above)
Comment (metacosm) [inline javaee/impl/src/main/java/org/jboss/forge/addon/javaee/rest/ui/RestNewEndpointCommandImpl.java]: Will reformat, sorry.

File: javaee/tests/src/test/java/org/jboss/forge/addon/javaee/jpa/ui/JPANewEntityCommandTest.java [hunk 3]
```diff
@@ -90,12 +88,12 @@ public void tearDown() throws Exception
    @Test
    public void checkCommandMetadata() throws Exception
    {
-      try (CommandController controller = uiTestHarness.createCommandController(JPANewEntityCommand.class,
+      try (CommandController controller = uiTestHarness.createCommandController(JPANewEntityCommandImpl.class,
```
Comment (gastaldi) [inline javaee/tests/src/test/java/org/jboss/forge/addon/javaee/jpa/ui/JPANewEntityCommandTest.java]: This test doesn't need to be changed

File: javaee/tests/src/test/java/org/jboss/forge/addon/javaee/rest/ui/RestNewEndpointCommandTest.java [hunk 4]
```diff
@@ -85,12 +80,12 @@ public void tearDown() throws Exception
    @Test
    public void checkCommandMetadata() throws Exception
    {
-      try (CommandController controller = uiTestHarness.createCommandController(RestNewEndpointCommand.class,
+      try (CommandController controller = uiTestHarness.createCommandController(RestNewEndpointCommandImpl.class,
```
Comment (gastaldi) [inline javaee/tests/src/test/java/org/jboss/forge/addon/javaee/rest/ui/RestNewEndpointCommandTest.java]: This test doesn't need to be changed

**Comment by gastaldi** [comment]
Retest this please

**Pull request merged**